### PR TITLE
Improve a11y context menu

### PR DESCRIPTION
**Pull request by yeion7**
## What kind of change does this PR introduce?

refactor

## What is the current behavior?

the context menu has a few accessibility issues 

## What is the new behavior?

use reakit for context menu, now the context menu is accessible 

## Checklist

- [ ] Documentation
- [ ] Testing 
- [ ] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by SaraVieira** [comment]
oh no

I just made a commit to the context menu :/

**Comment by yeion7** [comment]
oh no jaja 

is this https://github.com/codesandbox/codesandbox-client/pull/2586?

File: packages/app/src/app/components/ContextMenu/index.tsx [hunk 1]
```diff
@@ -20,7 +26,10 @@ type Item = ItemType | ItemType[];
 type ChildrenProps =
   | {
       childFunction: true;
-      children: (onContextMenu: OnContextMenu) => React.ReactNode;
+      children: (
+        onContextMenu: OnContextMenu,
+        disclosureProps: any
```
Comment (MichaelDeBoey) [inline packages/app/src/app/components/ContextMenu/index.tsx]: I think using `unknown` is a better choice.

```suggestion
        disclosureProps: unknown
```

File: packages/app/src/app/components/ContextMenu/index.tsx [hunk 2]
```diff
@@ -20,7 +26,10 @@ type Item = ItemType | ItemType[];
 type ChildrenProps =
```
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply packages/app/src/app/components/ContextMenu/index.tsx]: Maybe even delete `ChildrenProps` and integrate it in `Props`? 🤔

File: packages/app/src/app/components/ContextMenu/index.tsx [hunk 3]
```diff
@@ -34,219 +43,128 @@ type Props = {
...
+        {item.title}
+      </MenuItem>
+    );
   };
 
-  componentWillUnmount() {
-    this.unregisterListeners();
-    this.unmounted = true;
-  }
+  const onMenuEvent: OnContextMenu = event => {
```
[the remark this answers is not in the document]
Comment (yeion7) [reply packages/app/src/app/components/ContextMenu/index.tsx]: in this case, we pass the event [here](https://github.com/codesandbox/codesandbox-client/pull/2588/files/1584690b9abbd397971444b9a9128e9a1335b4f5#diff-9fd8cfad0b8a0a7f7831e0069b1cb6d8R134)

**Comment by yeion7** [comment]
@MichaelDeBoey thanks for all your comments, I've made the changes 👊

**Comment by CompuIves** [comment]
Wow, this is really cool, thanks for doing this @yeion7!!

I have two things I noticed:

1. It seems like the ContextMenu doesn't show up anymore at the cursor (when pressing right click in the top left of the card it still appears somewhere in the middle): 
image

2. I can't use the keyboard yet to select the items (whenever I right click I can't use tab/arrow keys to select items), I'm not sure if this PR wants to cover this.

**Comment by yeion7** [comment]
@CompuIves nice catch

* When it's open the menu with  `onContextMenu` event, I set a [new reference](https://github.com/codesandbox/codesandbox-client/pull/2588/files#diff-9fd8cfad0b8a0a7f7831e0069b1cb6d8R119) with the click position but in the sandbox card the reference  already set in the button and Reakit is taken the button ref, (I'll open an issue for this in reakit project)
* The problem with the list was I conditionally render  the menu but Reakit does not support it, because of this when the menu is toggle the focus is lost

**Comment by christianalfoni** [comment]
@CompuIves Just bumping you on this as everything seems to be okay now? Would be nice to bring in. Really awesome work @yeion7 !

**Comment by SaraVieira** [comment]
Looks good to me! Awesome work really

@CompuIves

**Comment by CompuIves** [comment]
Hmm, for me it doesn't work yet. When opening the context menu, it opens on a specific position, because of this I sometimes randomly click on an item (so I right click, it opens and *poof* it closes again and I download a zip). Otherwise it opens on another position than my mouse cursor. 

image

I'm wondering if it also would be possible to remove this border:

image

Since people know that there is a highlight since they've just opened it. Or we style it better.

**Pull request closed without merging**